molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/snpEff/SnpEffRunner.java

```diff
@@ -0,0 +1,366 @@
+
+	@SuppressWarnings("resource")
+	public Iterator<Entity> getSnpEffects(Iterator<Entity> source, final File inputVcf)
+	{
+		try
+		{
+			if (!source.hasNext()) return Iterators.emptyIterator();
+
+			// get meta data by peeking at the first entity (work-around for issue #4701)
+			PeekingIterator<Entity> peekingIterator = Iterators.peekingIterator(source);
+			EntityMetaData sourceEMD = peekingIterator.peek().getEntityMetaData();
+
+			List<String> params = Arrays.asList("-Xmx2g", getSnpEffPath(), "hg19", "-noStats", "-noLog", "-lof",
+					"-canon", "-ud", "0", "-spliceSiteSize", "5");
+			File outputVcf = jarRunner.runJar(NAME, params, inputVcf);
+
+			File snpEffOutputWithMetaData = addVcfMetaDataToOutputVcf(outputVcf);
+			VcfRepository repo = new VcfRepository(snpEffOutputWithMetaData, "SNPEFF_OUTPUT_VCF_" + inputVcf.getName());
+
+			SnpEffResultIterator snpEffResultIterator = new SnpEffResultIterator(repo.iterator());
+
+			return new Iterator<Entity>()
+			{
+				Iterator<Entity> sourceEntities = peekingIterator;
+				SnpEffResultIterator resultEntities = snpEffResultIterator;
+				LinkedList<Entity> effects = Lists.newLinkedList();
+
+				@Override
+				public boolean hasNext()
+				{
+					return (sourceEntities.hasNext() || !effects.isEmpty());
+				}
+
+				@Override
+				public Entity next()
+				{
+					if (effects.isEmpty())
+					{
+						// go to next source entity and get effects
+						Entity sourceEntity = sourceEntities.next();
+						String chromosome = sourceEntity.getString(VcfRepository.CHROM);
+						Long position = sourceEntity.getLong(VcfRepository.POS);
+
+						if (chromosome != null && position != null)
+						{
+							Entity snpEffEntity = resultEntities.get(chromosome, position);
+							if (snpEffEntity != null)
+							{
+								effects.addAll(getSnpEffectsFromSnpEffEntity(sourceEntity, snpEffEntity,
+										getOutputMetaData(sourceEMD)));
+							}
+							else
+							{
+								effects.add(getEmptyEffectsEntity(sourceEntity, getOutputMetaData(sourceEMD)));
+							}
+						}
+						else
+						{
+							effects.add(getEmptyEffectsEntity(sourceEntity, getOutputMetaData(sourceEMD)));
+						}
+					}
+					return effects.removeFirst();
+				}
+
+			};
+		}
+		catch (IOException e)
+		{
+			throw new UncheckedIOException(e);
+		}
+		catch (InterruptedException e)
+		{
+			throw new MolgenisDataException("Exception running SnpEff", e);
+		}
+	}
+
+	private Entity getEmptyEffectsEntity(Entity sourceEntity, EntityMetaData effectsEMD)
+	{
+		MapEntity effect = new MapEntity(effectsEMD);
+		effect.set(ID, idGenerator.generateId());
+		effect.set(VARIANT, sourceEntity);
+
+		return effect;
+	}
+
+	// ANN=G|intron_variant|MODIFIER|LOC101926913|LOC101926913|transcript|NR_110185.1|Noncoding|5/5|n.376+9526G>C||||||,G|non_coding_exon_variant|MODIFIER|LINC01124|LINC01124|transcript|NR_027433.1|Noncoding|1/1|n.590G>C||||||;
+	private List<Entity> getSnpEffectsFromSnpEffEntity(Entity sourceEntity, Entity snpEffEntity,
```

Comment:
private method from which the code is pretty straight forward, why do you think javadoc is necessary?